File: L17AdobeMedium/NumberOfLesserBeamsInBank.java

import java.util.*;

public class NumberOfLesserBeamsInBank {

    public int numberOfBeams(List<String> bank) {
        List<Integer> ans = new ArrayList<>();

        for (String row : bank) {
            int c = (int) row.chars().filter(ch -> ch == '1').count();
            if (c != 0) {
                ans.add(c);
            }
        }

        int sum = 0;
        if (ans.size() <= 1) {
            return 0;
        }

        for (int i = 0; i < ans.size() - 1; i++) { // Fixed loop condition
            sum += ans.get(i) * ans.get(i + 1); // Fixed incorrect index usage
        }

        return sum; // Moved return statement outside the loop
    }

    public static void main(String[] args) {
        List<String> bank = Arrays.asList(
                "00110",
                "10110",
                "01110",
                "11110"
        );

        NumberOfLesserBeamsInBank solution = new NumberOfLesserBeamsInBank();
        int result = solution.numberOfBeams(bank);

        System.out.println("Number of beams: " + result);
    }
}

File: L17AdobeMedium/NumberOfLesserBeamsInBank.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int numberOfBeams(vector<string>& bank) {
        vector<int> ans;
        
        // Step 1: Count the number of '1's in each row and store in `ans`
        for (int i = 0; i < bank.size(); i++) {
            int c = count(bank[i].begin(), bank[i].end(), '1');
            if (c != 0) {
                ans.push_back(c); // Only store rows with at least one '1'
            }
        }

        // Step 2: Calculate the total number of beams
        int sum = 0;
        for (size_t i = 0; i + 1 < ans.size(); i++) {
            sum += ans[i] * ans[i + 1]; // Multiply consecutive counts
        }
        
        return sum;
    }
};

int main() {
    vector<string> bank = {
        "00110",
        "10110",
        "01110",
        "11110"
    };

    Solution solution;
    int result = solution.numberOfBeams(bank);
    
    cout << "Number of beams: " << result << endl;
    
    return 0;
}
